Deprecation Warning in AudioDecoder.cc

## Environment
* OS Version: macOS Homebrew
* Source or binary build?  Source - targeting main

## Description
* Expected behavior: Builds without warning
* Actual behavior: Builds with warning

## Output

https://build.osrfoundation.org/job/ignition_common-ci-pr_any-homebrew-amd64/1734/clang/

```
 23%] Building CXX object graphics/src/CMakeFiles/gz-common6-graphics.dir/BVHLoader.cc.o
/Users/jenkins/workspace/ignition_common-ci-pr_any-homebrew-amd64/ign-common/av/src/AudioDecoder.cc:152:36: warning: 'channels' is deprecated [-Wdeprecated-declarations]
          this->dataPtr->codecCtx->channels;
                                   ^
/usr/local/Cellar/ffmpeg/5.1/include/libavcodec/avcodec.h:1005:5: note: 'channels' has been explicitly marked deprecated here
    attribute_deprecated
    ^
/usr/local/Cellar/ffmpeg/5.1/include/libavutil/attributes.h:100:49: note: expanded from macro 'attribute_deprecated'
#    define attribute_deprecated __attribute__((deprecated))
                                                ^
1 warning generated.
```

--- a/av/src/AudioDecoder.cc
+++ b/av/src/AudioDecoder.cc
@@ -149,7 +149,7 @@ bool AudioDecoder::Decode(uint8_t **_outBuffer, unsigned int *_outBufferSize)
         // decodedFrame->linesize[0].
         int size = decodedFrame->nb_samples *
           av_get_bytes_per_sample(this->dataPtr->codecCtx->sample_fmt) *
-          this->dataPtr->codecCtx->channels;
+          this->dataPtr->codecCtx->ch_layout.nb_channels;
 
         // Resize the audio buffer as necessary
         if (*_outBufferSize + size > maxBufferSize)

